src/components/Search/search-result.jsx

```diff
@@ -0,0 +1,106 @@
+import { Link } from 'gatsby';
+import { default as React } from 'react';
+import { Paragraph } from '@smallstep/step-ui';
+import { makeStyles } from '@material-ui/styles';
+import {
+  connectStateResults,
+  Highlight,
+  Hits,
+  Index,
+  Snippet,
+} from 'react-instantsearch-dom';
+import { Box, ListItem, ListItemText } from '@material-ui/core';
+
+const useStyles = makeStyles({
+  resultsBorder: {
+    borderColor: '#D3D3D3',
+    marginBottom: -16,
+    marginTop: -2,
+  },
+  hits: {
+    marginLeft: -30,
+    marginRight: 10,
+    '& mark ': {
+      backgroundColor: '#84A8FF',
+    },
+    '& ul ': {
+      listStyle: 'none',
+    },
+    '& a:-webkit-any-link ': {
+      textDecoration: 'none',
+      color: 'inherit',
+    },
+  },
+});
+
+const HitCount = connectStateResults(({ searchResults }) => {
+  const hitCount = searchResults && searchResults.nbHits;
+  const theme = useStyles();
+  return hitCount > 0 ? (
+    <Box borderBottom={1} textAlign={'right'} className={theme.resultsBorder}>
+      <Box mr={2}>
+        {hitCount} result{hitCount !== 1 ? `s` : ``}
+      </Box>
+    </Box>
+  ) : (
+    <Box
+      className={theme.resultsBorder}
+      minWidth={295}
+      textAlign={'left'}
+      ml={0.5}
+    >
+      There are no results found
+    </Box>
+  );
+});
+
+const PageHit = ({ hit }) => (
+  <Box mx={-2}>
+    <Link
+      to={
+        hit.title[0] === hit.title[0].toUpperCase()
```

Comment:
Confused. What is this check doing? We're determining URL based on whether the title is all caps?

Reply:
The slugs for the step cli commands are stored like this. The command step beta, for example, the slug is just '/beta' so we had to find a way to append `step-cli/reference/` to the slug. Since the titles of all the cli commands are lowercase this was our way of checking if the search result was a cli command or not.

Reply:
After playing around today and a lot the other day, I still can't seem to find a way to get a longer version of the slugs into Algolia. I was looking around on the GraphiQL and saw some other graphql queries that would grab the right slug but when I tried using them to talk to Algolia it would throw me an error from Algolia since it did not recognize certain attributes. 
There is an attribute that takes in the full file path but there would still need to be a bit of a work around to trim down all the unnecessary file path info.

I may just be missing some knowledge on Graphql in general or something but for me I could not find a good way to get better slugs into Algolia.

Reply:
Ok, that stinks, but I don't think it's worth holding up shipping. Once everything else is fully ready for review, let's go ahead and get a last look at it, then our team can come back around and see if we get to fixing this. I made #120 so we don't lose track. :)